src/python/pants/backend/python/util_rules/pex_cli.py

```diff
@@ -164,6 +167,10 @@ async def setup_pex_cli_process(
         "--tmpdir",
         tmpdir,
     ]
+
+    if request.concurrency_available > 0:
```

Comment:
I'm not sure this guard is needed. If there are 0 requirements and it's source only, then Pex does nothing in parallel anyhow.

Reply:
This is to avoid including the templated argument string if `request.concurrency_available == 0`, because then we would not template the variable at all. The intent there (not templating if no concurrency was claimed) was to avoid misuse, but it means that callers need a conditional like this.

Reply:
Yeah, but the assignment of the variable should probably say `len(requirements) if requirements or 1`

Reply:
> Yeah, but the assignment of the variable should probably say `len(requirements) if requirements or 1`

The `@rule` that consumes `PexCliProcess` (`setup_pex_cli_process`) does not have access to the requirement strings: they're available in `build_pex`. And there are multiple uses of `PexCliProcess`.

Reply:
Aha, ok.